python/ray/data/_internal/lazy_block_list.py

```diff
@@ -362,6 +362,31 @@ def __next__(self):
 
         return Iter()
 
+    def randomize_block_order(self, seed: Optional[int] = None):
+        """Randomizes the order of the blocks.
+
+        Args:
+            seed: Fix the random seed to use, otherwise one will be chosen
+                based on system randomness.
+        """
+        import random
+
+        if seed is not None:
+            random.seed(seed)
+
+        tasks_blocks_metadata = zip(
+            self._tasks,
+            self._block_partition_refs,
+            self._block_partition_meta_refs,
+        )
```

Comment:
Don't forget the cached metadata! https://github.com/ray-project/ray/blob/98c9303ca6c686dcb613c19c26dfad8f5ab57f83/python/ray/data/_internal/lazy_block_list.py#L67-L70

Reply:
🤔  should this be added to `split` as well?

Reply:
Yes they should, good catch! I'll open an issue for add to `split`, `split_by_bytes`, and `divide`.

Reply:
https://github.com/ray-project/ray/issues/25574